engine/time/src/main/java/io/deephaven/time/calendar/Calendar.java

```diff
@@ -1,280 +1,643 @@
- * Methods on Instant may not be precisely defined enough to return an Instant, e.g nextDay(). In these cases, the
- * method will return a String as discussed above.
- *
- *
- * To maintain consistency, each calendar has two fields: a name, and a time zone. A calendar with the same schedule but
- * a different time zone is considered a different calendar.
- *
- *
- * Frequently, the default implementation for methods on Instants is to call the corresponding method on a String with
- * {@code DateTimeUtils.formatDate}. This can be slower than methods written explicitly for DateTimes. If performance is
- * an issue, consider overriding these methods with other behavior.
+ * <p>
+ * A calendar is associated with a specific time zone.
+ * <p>
+ * Date strings must be in a format that can be parsed by {@code DateTimeUtils#parseDate}. Methods that accept strings
+ * can be slower than methods written explicitly for {@code Instant}, {@code ZonedDateTime}, or {@code LocalDate}.
  */
-public interface Calendar {
+public class Calendar {
+
+    private final String name;
+    private final String description;
+    private final ZoneId timeZone;
+
+    // region Constructors
+
+    /**
+     * Creates a new calendar.
+     *
+     * @param name calendar name.
+     * @param description calendar description.
+     * @param timeZone calendar time zone.
+     * @throws RequirementFailure if any parameter is {@code null}
+     */
+    public Calendar(final String name, final String description, final ZoneId timeZone) {
+        this.name = Require.neqNull(name, "name");
+        this.description = Require.neqNull(description, "description");
+        this.timeZone = Require.neqNull(timeZone, "timeZone");
+    }
+
+    // endregion
+
+    // region Getters
 
     /**
      * Gets the name of the calendar.
      *
      * @return the name of the calendar
      */
-    String name();
+    public String name() {
+        return name;
+    }
 
     /**
-     * Gets the current date.
+     * Gets the description of the calendar.
      *
-     * @return the current day
+     * @return the description of the calendar
      */
-    default String currentDay() {
-        return DateTimeUtils.today(timeZone());
+    public String description() {
+        return description;
     }
 
     /**
-     * Gets yesterday's date.
+     * Gets the timezone of the calendar.
+     *
+     * @return the time zone of the calendar
+     */
+    public ZoneId timeZone() {
+        return timeZone;
+    }
+
+    @Override
+    public String toString() {
+        return "Calendar{" +
+                "name='" + name + '\'' +
+                ", description='" + description + '\'' +
+                ", timeZone=" + timeZone +
+                '}';
+    }
+
+    // endregion
+
+    // region Now
+
+    /**
+     * The current date for the calendar.
+     *
+     * @return current date
+     */
+    public LocalDate calendarDate() {
+        return DateTimeUtils.todayLocalDate(timeZone());
```

Comment:
It's important context, but we are depending on the `clock` state in `DateTimeUtils`. I'm not the biggest fan of the static state in `DateTimeUtils` in-and-of itself, but I wonder if we should be persisting that stateful dependency here? Should `clock` be a member of `Calendar`?

Reply:
I'm not fully following this set of comments.

`DateTimeUtils` is the foundational layer.  `Calendar` is built on top of `DateTimeUtils`.  `BusinessCalendar` is built on top of `Calendar`.  If `Calendar` has its own clock, separate from `DateTimeUtils`, it is just going to lead to user nightmare scenarios since we can't tell the user to just change the clock in a single place.  Instead, they will end up with an inconsistent view of the world where they need to know multiple places they need to update to achieve consistency.